### PR TITLE
QA-2282: Add app-version output

### PR DESCRIPTION
**Pull request by ichengchang**
Ticket: [QA-2282](https://broadworkbench.atlassian.net/browse/QA-2282)

Use the following logic to determine app versioning for contract tests
If the current commit hash for tag is the same as the current commit hash, 

- app version of published contract = tag from bumper output, otherwise
- app version of published contract  = new_tag from bumper output

<Put notes here to help reviewer understand this PR>

---

**PR checklist**

- [ ] Include the JIRA issue number in the PR description and title
- [ ] Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] If you changed anything in `model/`, then you should [publish a new official `rawls-model`](https://github.com/broadinstitute/rawls/blob/develop/README.md#publish-rawls-model) and update `rawls-model` in [Orchestration's dependencies](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/project/Dependencies.scala).
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green, including CI tests
- [ ] **Squash commits and merge** to develop (branches are automatically deleted after merging)
- [ ] Inform other teams of any substantial changes via Slack and/or email

[QA-2282]: https://broadworkbench.atlassian.net/browse/QA-2282?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by calypsomatic** [comment]
> Use the following logic to determine app versioning for contract tests If the current commit hash for tag is the same as the current commit hash,
> 
This check happens in the bumper workflow?

**Review by calypsomatic — APPROVED** [review]

File: .github/workflows/tag.yml [hunk 1]
```diff
@@ -60,7 +59,13 @@ jobs:
         DRY_RUN: ${{ inputs.dry-run }}
         RELEASE_BRANCHES: ${{ inputs.release-branches }}
         WITH_V: true
-    - name: Echo generated tag to console
-      if: ${{ inputs.print-tag == 'true' }}
+    - name: Output app version
+      id: output-version
       run: |
-        echo "Newly created version tag: '${{ steps.tag.outputs.new_tag }}'"
+        if [[ -z "${{ steps.tag.outputs.new_tag }}" ]]; then
```
Comment (jyang-broad) [inline .github/workflows/tag.yml:66]: Context Request: Add additional conctext (jira ticket?) that explains this is for possible race-condition deconfliction.

**Review by jyang-broad — APPROVED** [review]
looks good! now compares tag + new-tag into app-version which is the "correct" version of pact contracts to test.

File: .github/workflows/tag.yml (hunk 1 above)
Comment (ichengchang) [reply .github/workflows/tag.yml:66]: Added ticket number as contextual comment

**Comment by ichengchang** [comment]
> > Use the following logic to determine app versioning for contract tests If the current commit hash for tag is the same as the current commit hash,
> 
> This check happens in the bumper workflow?

Yes this check happens in the bumper workflow, as shown below

[https://github.com/DataBiosphere/github-actions/blob/master/actions/bumper/entrypoint.sh#L76-L88](https://github.com/DataBiosphere/github-actions/blob/master/actions/bumper/entrypoint.sh#L76-L88)

```
# get current commit hash for tag
# on the initial use, this shows usage, because $tag is empty. The logic below still works.
tag_commit=$(git rev-list -n 1 $tag)

# get current commit hash
commit=$(git rev-parse HEAD)

if [ "$tag_commit" == "$commit" ]; then
    echo "No new commits since previous tag. Skipping..."
    echo tag=$tag >> $GITHUB_OUTPUT
    exit 0
fi
```

**Pull request merged**